### ISSUE TITLE
decomp_kinship crashes R if ncol=nrow=0

### ISSUE THREAD

**Issue by byandell** [rqtl/qtl2scan#94]
If get_common_ids returns 0-length vector ind2keep, subsequent call to
decomp_kinship(kinship[ind2keep,ind2keep])
crashes R when Rcpp code called. Brian

**Comment by kbroman** [comment]
Could you give a specific example? I can see a few places where my handling of pre-decomposed kinship matrices could be improved, but it should be that if `length(ind2keep)<=2` things stop with an error.

**Comment by kbroman** [comment]
@byandell, I cleaned up some of the code dealing with kinship; does it fix the problem? if you give me an example, I'd add it as a test.

**Comment by byandell** [comment]
This is an example that broke the code. I just downloaded your updated code, but it still crashes for example below.

You check in various routines that ind2keep is at least 2 before calling, but I did not notice that. The actual situation was using covariates that had no overlap with the phenotype, hence get_common_ids() returned vector of length 0. Putting a simple check that ncol(kinship) >= 2 in decomp_kinship would address this, I think.
Here is simplified code (use your favorite kinship of course:

library(qtl2scan)
datapath <- "yourpath"
kinship <- readRDS(file.path(datapath, "kinship.rds"))[["17"]]
ind2keep <- integer()
K <- qtl2scan::decomp_kinship(kinship[ind2keep, ind2keep])
